src/domain/dataset_bert.py

```diff
@@ -35,22 +50,65 @@ def _load_dataset(self, train_dir, test_dir, val_dir, cache_dir="./cache_dir"):
                             data_files={"train": train_files, "test": test_files, "val": val_files},
                             cache_dir=cache_dir)
         
-    def encode(self, document, tokenizer): 
+    def encode_document(self, document, tokenizer): 
         """Utility method used to preprocess and tokenize the data
         
         Args:
-            document (list): list of lists of sentences to preprocess and tokenize
+            document (list): list of sentences to preprocess and tokenize
             tokenizer (BertTokenizerFast): object method used to preprocess and tokenize 
 
         Return:
             dict: dictionary with keys `input_ids`, `token_type_ids` and `attention_mask`
         """
-        return tokenizer(document, 
-                        max_length=80,
-                        add_special_tokens=True,
-                        padding=True,
-                        truncation=True,
-                        return_tensors='pt')
+
+        #concat sentences into document
+        result_ ={'token_ids':[],
+                'token_type_ids':[],
+                'mark':[],
+                'segs':[],
+                'clss':[0],
+                'mark_clss':[True]}
+        seg=1
+        for sentence in document:
+            seg=1-seg
```

Comment:
Truc perso, mais j'aurais plus utilisé un enumerate dans la boule for plus haut et utilisé 

``seg = i % 2``

C'est un peu plus clair que ça alterne entre 0 et 1.

Reply:
Corrigé! ;) C'est vrai que c'est un peu plus pertinent.